aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -160,20 +160,28 @@ Tensor& eye_out_cpu(Tensor& result, int64_t n, int64_t m) {
   return result;
 }
 
+Tensor eye_like(const Tensor& self) {
+  AT_CHECK(self.dim() == 2, "Expected input to be two-dimensional, instead got ",
+           self.dim(), " dimensions");
+  return native::eye(self.size(0), self.size(1), self.options());
```

Comment:
Does this work for sparse? (What is the error message, if there is one?)

Reply:
It doesn't work for sparse. The error message falls back to the one printed by `eye_out`, but I have added it here too.